### PR TITLE
Milestone 2:  Initialize Redux store and connect given components to state

### PR DESCRIPTION
**Pull request by moinkhanif**
This PL focuses on creating:
- Redux Store
  - `reducers`
  - `actions`
  - `containers`
  - connecting redux store

### CODE REVIEW

**Comment by sumancrest0001** [comment]
### Status: Required changes :gear:

**Hello @moinkhanif**, You did a good job in implementing requirements from the milestone-2. We are almost done with this milestone. There are minor changes required before moving ahead. Let's have a look at them one by one.

File: src/components/BooksList/BooksList.component.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+BookList.defaultProps = {
+  books: {
+    id: 'Undefined',
+    title: 'Undefined',
+    category: 'Undefined',
+  },
+};
+
```
Comment (sumancrest0001) [inline src/components/BooksList/BooksList.component.jsx]: **[Required]** Is it required to pass these default props? Consider a book with an Undefined id, title, and category. I think it doesn't make sense.

File: src/reducers/books.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      };
+    case REMOVE_BOOK:
+      return {
+        ...state,
+        REMOVE_BOOK: true,
+      };
+    default:
+      return state;
+  }
+};
```
Comment (sumancrest0001) [inline src/reducers/books.js]: **[Required]** The bookReducer function is not correct. I think you are mismatching actions and reducer. For example, We get {[existingbookArray, newbook: book]} when CREATE_BOOK is dispatched. In the REMOVE_BOOK you are not passing **id** the book that needs to be deleted. How does it know which book to remove? We can pass an id as a payload and filter the book from bookArray. [Reference](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/ReducxReducers/CreateAndRemoveReducers.png)

File: src/components/BooksForm/BooksForm.component.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const BOOK_CATEGORY = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
```
Comment (sumancrest0001) [inline src/components/BooksForm/BooksForm.component.jsx]: **[Required]** BooksForm component should be in the container directory.

**Review by sumancrest0001 — CHANGES_REQUESTED** [review]
**[Optional]** There are a couple of console errors. You can fix them.

**Please make the requested changes and request for another code review. 
If you have any queries regarding this review, message me on Slack
Suman Shrestha**

**Happy coding !!!**

**Comment by moinkhanif** [comment]
Hey @sumancrest0001 , thank you for a detailed review. I have implemented the following changes based on your feedback:

- Set the props to `required` so as to avoid the default props issue.
- Fixed all console errors. Turns out they were all due to incorrect PropTypes.
- Converted `CREATE_BOOK` and `REMOVE_BOOK` into functions that takes in `book` and current state as arguments and return a new State. I also made sure not to mutate the previous state in the process.
- Moved `BooksForm` to the container folder.

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: 
Hi @moinkhanif :hand: 
Nice work implementing the requested changes.
Good luck with the next PR.

Happy coding :computer: 
**Sergio Torres**

**Pull request merged**